homeworks/class3/routes/cats.js

```diff
@@ -0,0 +1,91 @@
+var catData = require('../cat-creator');
+var Cat = require('mongoose').model('Cat');
+
+module.exports.showAll = function (req, res) {
+	var pageData = {
+		'page-title' : "All",
+		'page-header': "All Cats",
+		'page-subheader': "Sorted by Age"
+	};
+
+	// Get Cats from DB
+	Cat.find()
+	   .sort({age: 1})
+	   .exec( function(err, cats) {
+			if (err) {
+				pageData.error = "Failed to locate cats!";
```

Comment:
This is a completely solid way to handle errors (showing the user) but the other (more traditional) is to send a 500 error status